File: app/src/main/java/ua/com/foxminded/locationtrackera/util/SafeNavigation.java
package ua.com.foxminded.locationtrackera.util;

import android.view.View;

import androidx.annotation.IdRes;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

public class SafeNavigation {

    public static void navigate(View view, @IdRes int destination, @IdRes int action) {
        final NavController controller = Navigation.findNavController(view);
        if (controller.getCurrentDestination().getId() == destination) {
            controller.navigate(action);
        }
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/util/SafeNavigation.kt
package ua.com.foxminded.locationtrackera.util

import android.view.View
import androidx.annotation.IdRes
import androidx.navigation.Navigation.findNavController

object SafeNavigation {

    fun navigate(view: View, @IdRes destination: Int, @IdRes action: Int) {
        val controller = findNavController(view)
        if (controller.currentDestination!!.id == destination) {
            controller.navigate(action)
        }
    }
}